src/computational_graph/graph.jl

```diff
@@ -280,13 +282,22 @@ julia> g.subgraphs
 3:ϕ(3)|ϕ(6)=0.0
 ```
 """
-function feynman_diagram(vertices::Vector{OperatorProduct}, topology::Vector{Vector{Int}};
-    external=[], factor=one(_dtype.factor), weight=zero(_dtype.weight), name="", type=:generic)
-
-    operators = [o for v in vertices for o in v.operators]
+function feynman_diagram(vertices::AbstractVector,
+    topology::Vector{Vector{Int}};
+    external=[], factor=one(_dtype.factor),
+    weight=zero(_dtype.weight),
+    name="", type=:generic)
+
+    # convert all Graphs to vertices by grouping all its external legs as a vertex
+    converted_vertices = [_extract_vertex(typeof(v), v) for v in vertices]
+    # collect graphs in vertices as subdiagrams
+    subdiagrams = [v for v in vertices if isa(v, Graph)]
+
+    operators = [o for v in converted_vertices for o in v.operators]
     permutation = collect(Iterators.flatten(topology))
     ind_ops = collect(eachindex(operators))
 
+
     @assert length(unique(permutation)) == length(permutation) # no repeated index
     @assert length(unique(external)) == length(external) # no repeated index
     @assert Set(permutation) == Set(ind_ops) # permutation must exhaust all operators
```

Comment:
This assertion should be removed since permutation will not exhaust all operators of converted vertices.

Reply:
why? the internal vertices of sub-diagrams are not passed to the main diagram, only the real external legs are converted to vertices in the main diagram, and all those operators should somehow be connected.